exploration/iri_seasons.md

```diff
@@ -0,0 +1,208 @@
+```
+
+```python
+stats = utils.load_iri_inseason_stats()
+stats["rel_month1"] = stats["rel_month1"].apply(
+    lambda x: x if x < 13 else x - 12
+)
+
+# calculate fraction of pixels in AOI that might have a season
+# i.e., ones that don't have an error code ( > 250) in ASAP rasters
+
+# note: must do as float, otherwise compute_raster_stats doesn't work
+s_pos_da = xr.where(utils.load_asap_inseason("dekad", 1) < 251, 1.0, 0.0)
+s_pos = s_pos_da.oap.compute_raster_stats(aoi, feature_col="ADM0_CODE")
+s_pos = s_pos.rename(
+    columns={"sum": "possible_season_pixels", "count": "total_pixels"}
+)
+
+# fraction with possible season is close to 100% for BFA, but less for TCD and NER
+display(s_pos)
+
+stats = stats.merge(
+    s_pos[["ADM0_CODE", "possible_season_pixels", "total_pixels"]],
+    on="ADM0_CODE",
+)
+stats["frac_reporting"] = stats["count"] / stats["possible_season_pixels"]
+display(stats)
+```
+
+## Selecting relevant dates
+
+We can select relevant dates either by:
+
+- Just picking which months we care about
+  - Based on previous frameworks this would be Jun - Oct
+- Only monitoring once the fraction of "reporting" pixels
+goes above a certain level, say 50%
+
+```python
+# check typical values by leadtime, month
+
+agg_cols = ["L", "rel_month1"]
+plot_cols = ["30quant", "frac_reporting"]
+
+for agg_col in agg_cols:
+    year_group = stats.groupby(["F_year", agg_col])[plot_cols].mean()
+    dff = year_group.groupby(agg_col)[plot_cols].agg(["max", "mean", "min"])
+    for plot_col in plot_cols:
+        fig, ax = plt.subplots()
+        dff[plot_col].plot(ax=ax)
+        ax.set_ylabel(plot_col)
+
+# note that rel_month1 is the first relevant month of the forecast
+
+# things are pretty stable by leadtime, except fewer tend to be reporting
+# at L=4
+
+# we see that if we do not select which months we care about,
+# we will often end up triggering in October
+```
+
+Unfortunately `frac_reporting` is still pretty high in September.
+This means that we could trigger for a forecast relevant for Sep-Oct-Noc,
+which doesn't seem right.
+
+Basically this means that we need to either:
+
+- Threshold based on how many dekads within the trimester each pixel is in season
+  - We are currently just using `any()` (i.e. the trimester is inseason
+if any of its dekads are inseason)
+- Use ECMWF which forecasts monthly instead of trimesterially
+
+Here we are doing the land area fraction as `pixels_above_threshold / pixels_reporting`,
+but this might not make sense. Based on previous frameworks, the analogue
+here would be more like `pixels_above_theshold / pixels_with_possible_season`.
+That way the denominator stays the same.
+
+```python
+# plot return period for different methods of selecting relevant dates
+
+frac_reporting_threshold = 0.5
+rel_months = [6, 7, 8]
+total_months = [x for x in range(min(rel_months), max(rel_months) + 3)]
+plot_quants = [10, 50, 90]
+plot_quant_cols = [f"{100 - x}quant" for x in percentiles if x in plot_quants]
```

Comment:
`percentiles` is undefined in this notebook.

Reply:
As is `max_per_year`

Reply:
And `percetile_cols`

Reply:
My bad! 07fb0ee8a9cb4d969cdaf356e5592c319212c81c